File: ConfiEscola/ClientApp/src/app/viewModels/alunoViewModel.ts
import { historicoEscolarViewModel } from "./historicoEscolarViewModel";

export class alunoViewModel{
    id: number;
    nome: string;
    sobrenome: string;
    email: string;
    dataNascimento:string;
    escolaridadeId:number;
    historicoEscolar: historicoEscolarViewModel

    constructor(data: alunoViewModel)
    {   
      this.id=data.id;   
      this.dataNascimento=data.dataNascimento;
      this.email=data.email;
      this.escolaridadeId = data.escolaridadeId;
      this.historicoEscolar=data.historicoEscolar;
      this.nome=data.nome;
      this.sobrenome=data.sobrenome;
    }
}

File: ConfiEscola.Application/ViewModels/AlunoViewModel.cs
using ConfiEscola.Domain.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConfiEscola.Application.ViewModels
{
    public class AlunoViewModel
    {
        public int Id { get; set; }
        public string Nome { get;  set; }
        public string Sobrenome { get; set; }
        public string Email { get; set; }
        public DateTime DataNascimento { get; set; }
        public int EscolaridadeId { get; set; }
        public HistoricoEscolarViewModel HistoricoEscolar { get; set; }
    }
}
